src/components/plots/ElbowPlot.jsx

```diff
@@ -122,25 +133,37 @@ const ElbowPlot = (props) => {
     },
   });
 
+  if (plotSpec) {
+    return (
+      <center>
+        <Vega data={{ plotData }} spec={plotSpec} renderer='canvas' actions={actions} />
+      </center>
+    );
+  }
+
   return (
-    <center>
-      <Vega data={{ plotData }} spec={generateSpec()} renderer='canvas' actions={actions} />
-    </center>
+    <PlatformError
+      description='There is no data to display. Please run the filter again.'
+      actionable={false}
+      reason={' '}
+    />
```

Comment:
This should probably be swapped around. Show the error first if no plot spec, and then render the actual plot otherwise.